internal/storage/connectors.go

```diff
@@ -31,6 +46,45 @@ type connector struct {
 	DecryptedConfig json.RawMessage `bun:"decrypted_config,scanonly"`
 }
 
+func (s *store) ListenConnectorsChanges(ctx context.Context, handlers HandlerConnectorsChanges) error {
+	conn, err := s.db.Conn(ctx)
+	if err != nil {
+		return errors.Wrap(err, "cannot get connection")
+	}
+	if err := conn.Raw(func(driverConn any) error {
+		listener := pgxlisten.Listener{
+			Connect: func(ctx context.Context) (*pgx.Conn, error) {
+				return pgx.Connect(ctx, driverConn.(*stdlib.Conn).Conn().Config().ConnString())
+			},
+		}
+		listener.Handle("connectors", pgxlisten.HandlerFunc(func(ctx context.Context, notification *pgconn.Notification, conn *pgx.Conn) error {
+			for prefix, handler := range handlers {
+				if strings.HasPrefix(notification.Payload, string(prefix)) {
+					return handler(
+						ctx,
+						models.MustConnectorIDFromString(strings.TrimPrefix(notification.Payload, string(prefix))),
```

Comment:
Are you choosing the use a function that panics in the listener on purpose? Although I get that we should never receive a notification for an unknown connector ID, and shutting down the container so that we restart the container with a fresh slate sounds reasonable in the case of such an unlikely event, I'm not so sure that panicking is a great idea.

Since we have some stateful workloads which we wait for during graceful shutdown, I kind of feel like adding potential to panic makes the entire application more fragile / introduces more potential for creating weird states if certain jobs get interrupted.

Reply:
the MustConnectorIDFromString implies that the id received is of a connector ID form, and that will be always the case since the notify from the database send an ID. It's impossible to panic here

Reply:
I would prefer to say "highly unlikely" rather than impossible 😁. But the presence of the wrapping `strings.HasPrefix()` filtering out any unexpected input is probably defensive enough as it is 👍